backend/api_test.py

```diff
@@ -1,19 +1,27 @@
 from fastapi.testclient import TestClient
-from api import app, healthy_response, error_message
+from api import app, healthy_response, unhealthy_neo4j_response, chat_fail_response
 
 client = TestClient(app)
 utterance = "Hello there"
+expected_message = "Hello to you too! From InferGPT"
 
-
-def test_health_check_response():
+def test_health_check_response_healthy():
     response = client.get("/health")
 
     assert response.status_code == 200
     assert response.json() == healthy_response

```

Comment:
The failing test is `test_health_check_response`. The test results in real calls to `test_connection()` causing `driver.verify_connectivity()` to make real network calls. As a result, the test only passes if neo4j is running (and using the URL/credentials provided in the `.env` file).

We should avoid real network calls in a unit tests. In other tests within `api.py` we're mocking `api.test_connection`,  so I think we need to add the same mocking behaviour to this healthy response test. This makes our test consistent with the other tests and ensures we're only testing the logic in `api.py` (as well as ensuring the test passes):

```
def test_health_check_response_healthy(mocker):

    mock_connection_test = mocker.patch("api.test_connection", return_value=True)

    response = client.get("/health")

    mock_connection_test.assert_called()

    assert response.status_code == 200
    assert response.json() == healthy_response
```

Reply:
Yes, no you're right. Happy to fix the tests to uncouple them from the service running here 👍

Reply:
I have mocked out the `test_connection()` call in the latest commit

![image](https://github.com/WaitThatShouldntWork/InferGPT/assets/48159708/bcb2c23a-45ff-4b40-a480-18bf9aa98209)